common/djangoapps/student/roles.py

```diff
@@ -49,7 +49,7 @@ def has_role(self, role, course_id, org):
         """
         return any(
             access_role.role == role and
-            access_role.course_id == course_id and
+            unicode(access_role.course_id) == unicode(course_id) and
```

Comment:
Because code was not able to compare ``ccxLocator`` objects though it was able to compare ``courseLocator``.
In sum when this line was comparing ccx keys it always give false even both key are match. Thats why I am deserialising keys and comparing them as strings.